pkg/proxy/metaproxier/meta_proxier.go

```diff
@@ -66,10 +160,25 @@ func (proxier *metaProxier) OnServiceAdd(service *v1.Service) {
 	if utilproxy.ShouldSkipService(service) {
 		return
 	}
-	if utilnet.IsIPv6String(service.Spec.ClusterIP) {
-		proxier.ipv6Proxier.OnServiceAdd(service)
-	} else {
-		proxier.ipv4Proxier.OnServiceAdd(service)
+
+	// we need to force a pre port sync before we call actuals. if actual raced
+	// to lock a port that we hold (for a service that was recently deleted)
+	// it will fail since we are holding it. so we sync our ports then call
+	// let actual know about the service update
+	// note: we do release port in a lazy way (no forced sync on delete)
+	proxier.addService(service)
+	proxier.syncPorts()
+
+	// this allows skew between new proxy and old apiserver
```

Comment:
Same question here about version skew, do we need to support new kube-proxy against old kube-apiserver? I don't think this is supported in the version skew policy.

Reply:
answered somewhere else.